pkg/controller/sync/version/cluster.go

```diff
@@ -65,18 +71,28 @@ func (a *clusterVersionAdapter) SetStatus(obj pkgruntime.Object, status *fedv1a1
 
 func (a *clusterVersionAdapter) Create(obj pkgruntime.Object) (pkgruntime.Object, error) {
 	version := obj.(*fedv1a1.ClusterPropagatedVersion)
-	return a.client.ClusterPropagatedVersions().Create(version)
+	err := a.client.Create(context.TODO(), version)
+	if err != nil {
+		return nil, err
+	}
+	return version, nil
 }
 
 func (a *clusterVersionAdapter) Get(qualifiedName util.QualifiedName) (pkgruntime.Object, error) {
-	return a.client.ClusterPropagatedVersions().Get(qualifiedName.Name, metav1.GetOptions{})
-}
-
-func (a *clusterVersionAdapter) Delete(qualifiedName util.QualifiedName) error {
-	return a.client.ClusterPropagatedVersions().Delete(qualifiedName.Name, nil)
+	objectKey := client.ObjectKey{Name: qualifiedName.Name}
```

Comment:
(No action required) I think `QualifiedName` is useful in the way it supports stringifying cleanly for both namespaced and non-namespaced resources.   Maybe it makes sense to add methods and helpers that convert to `client.ObjectKey` since that will be used in all the client calls?